core/arch/aarch64/proc.c

```diff
@@ -187,98 +184,73 @@ proc_init_arch(void)
         LOG_FEATURE(FEATURE_RNG);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64ISAR1_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64isar1);
+            cpu_info.features.isa_features[AA64ISAR1]);
         LOG_FEATURE(FEATURE_DPB);
         LOG_FEATURE(FEATURE_DPB2);
         LOG_FEATURE(FEATURE_JSCVT);
+        LOG_FEATURE(FEATURE_PAUTH);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64PFR0_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64pfr0);
+            cpu_info.features.isa_features[AA64PFR0]);
         LOG_FEATURE(FEATURE_FP16);
         LOG_FEATURE(FEATURE_RAS);
         LOG_FEATURE(FEATURE_SVE);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64MMFR1_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64mmfr1);
+            cpu_info.features.isa_features[AA64MMFR1]);
         LOG_FEATURE(FEATURE_LOR);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64DFR0_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64dfr0);
+            cpu_info.features.isa_features[AA64DFR0]);
         LOG_FEATURE(FEATURE_SPE);
-        LOG_FEATURE(FEATURE_PAUTH);
         LOG_FEATURE(FEATURE_LRCPC);
         LOG_FEATURE(FEATURE_LRCPC2);
         LOG_FEATURE(FEATURE_FRINTTS);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64ZFR0_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64zfr0);
+            cpu_info.features.isa_features[AA64ZFR0]);
         LOG_FEATURE(FEATURE_BF16);
         LOG_FEATURE(FEATURE_I8MM);
         LOG_FEATURE(FEATURE_F64MM);
 
         LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64PFR1_EL1 = 0x%016lx\n",
-            cpu_info.features.flags_aa64pfr1);
+            cpu_info.features.isa_features[AA64PFR1]);
         LOG_FEATURE(FEATURE_MTE);
         LOG_FEATURE(FEATURE_BTI);
+
+        LOG(GLOBAL, LOG_TOP, 1, "Processor features:\n ID_AA64ISAR2_EL1 = 0x%016lx\n",
+            cpu_info.features.isa_features[AA64ISAR2]);
+        LOG_FEATURE(FEATURE_PAUTH2);
     });
 #    endif
 #endif
 }
 
-#define GET_FEAT_REG(FEATURE) (feature_reg_idx_t)((((ushort)FEATURE) & 0x7F00) >> 8)
+#define GET_FEAT_REG(FEATURE) (feature_reg_idx_t)((((ushort)FEATURE) & 0x3F00) >> 8)
```

Comment:
Where is this 0x3F00 coming from?  Is this from DEF_FEAT?  How about a comment showing the bit #s and what each one is.

Reply:
I have added a breakdown of the bits in proc_api.h